### PR TITLE
feat: feature gate DefaultPodTopologySpread is GA

### PR DESCRIPTION
**Pull request by kerthcet**
Signed-off-by: kerthcet <kerthcet@gmail.com>

Part of https://github.com/kubernetes/enhancements/issues/1258

### CODE REVIEW

**Comment by kerthcet** [comment]
cc @alculquicondor

File: content/en/docs/concepts/workloads/pods/pod-topology-spread-constraints.md [hunk 1]
```diff
@@ -328,11 +318,9 @@ using default constraints for `PodTopologySpread`.
 
 #### Internal default constraints
```
Comment (sftim) [inline content/en/docs/concepts/workloads/pods/pod-topology-spread-constraints.md:319]:
```suggestion
### Built-in default constraints {#internal-default-constraints}
```
or
```suggestion
### Automatic default constraints {#internal-default-constraints}
```

File: content/en/docs/concepts/workloads/pods/pod-topology-spread-constraints.md [hunk 2]
```diff
@@ -328,11 +318,9 @@ using default constraints for `PodTopologySpread`.
...
 
-{{< feature-state for_k8s_version="v1.20" state="beta" >}}
+{{< feature-state for_k8s_version="v1.24" state="stable" >}}
 
-With the `DefaultPodTopologySpread` feature gate, enabled by default, the
-legacy `SelectorSpread` plugin is disabled.
-kube-scheduler uses the following default topology constraints for the
+Kube-scheduler uses the following default topology constraints for the
 `PodTopologySpread` plugin configuration:
```
Comment (sftim) [inline content/en/docs/concepts/workloads/pods/pod-topology-spread-constraints.md:324]:
```suggestion
If you don't configure any cluster-level default constraints for pod topology spreading,
then then kube-scheduler acts as you specified the the following default topology constraints:
```

**Review by sftim — COMMENTED** [review]
Should we also revise this text:
```markdown
The score produced by default scheduling constraints might conflict with the
score produced by the
[`SelectorSpread` plugin](/docs/reference/scheduling/config/#scheduling-plugins).
It is recommended that you disable this plugin in the scheduling profile when
using default constraints for `PodTopologySpread`.
```

eg to highlight that `SelectorSpread` is disabled by default, so the advice is really “don't enable this”

File: content/en/docs/concepts/workloads/pods/pod-topology-spread-constraints.md (hunk 1 above)
Comment (alculquicondor) [reply content/en/docs/concepts/workloads/pods/pod-topology-spread-constraints.md:319]: Built-in sgtm

File: content/en/docs/concepts/workloads/pods/pod-topology-spread-constraints.md (hunk 2 above)
Comment (alculquicondor) [reply content/en/docs/concepts/workloads/pods/pod-topology-spread-constraints.md:324]: `as if you`? I'm not a native speaker, so feel free to disregard if I made a mistake :)

Comment (sftim) [reply content/en/docs/concepts/workloads/pods/pod-topology-spread-constraints.md:324]: “as if you” is correct; sorry!

Comment (kerthcet) [reply content/en/docs/concepts/workloads/pods/pod-topology-spread-constraints.md]: thanks. updated.

File: content/en/docs/concepts/workloads/pods/pod-topology-spread-constraints.md [hunk 3]
```diff
@@ -319,21 +309,17 @@ profiles:
...
-[`SelectorSpread` plugin](/docs/reference/scheduling/config/#scheduling-plugins).
-It is recommended that you disable this plugin in the scheduling profile when
-using default constraints for `PodTopologySpread`.
+[`SelectorSpread` plugin](/docs/reference/scheduling/config/#scheduling-plugins)
+is disabled by default. It's recommended to use `PodTopologySpread` to achieve similar
+behavior.
 {{< /note >}}
 
-#### Internal default constraints
+#### Build-in default constraints
```
Comment (kbhawkey) [inline content/en/docs/concepts/workloads/pods/pod-topology-spread-constraints.md:317]: nit: Built-in?

Comment (kerthcet) [reply content/en/docs/concepts/workloads/pods/pod-topology-spread-constraints.md:317]: Aha..., good cache. Thanks.

File: content/en/docs/concepts/workloads/pods/pod-topology-spread-constraints.md [hunk 4]
```diff
@@ -319,21 +309,17 @@ profiles:
...
-[`SelectorSpread` plugin](/docs/reference/scheduling/config/#scheduling-plugins).
-It is recommended that you disable this plugin in the scheduling profile when
-using default constraints for `PodTopologySpread`.
+[`SelectorSpread` plugin](/docs/reference/scheduling/config/#scheduling-plugins)
+is disabled by default. It's recommended to use `PodTopologySpread` to achieve similar
+behavior.
 {{< /note >}}
 
-#### Internal default constraints
+#### Built-in default constraints
```
Comment (sftim) [inline content/en/docs/concepts/workloads/pods/pod-topology-spread-constraints.md:317]: We can preserve existing bookmarks etc
```suggestion
#### Built-in default constraints {#internal-default-constraints}
```

Comment (kerthcet) [reply content/en/docs/concepts/workloads/pods/pod-topology-spread-constraints.md]: updated.

**Pull request merged**